rs/types/management_canister_types/src/lib.rs

```diff
@@ -808,6 +835,174 @@ impl From<pb_canister_state_bits::LogVisibility> for LogVisibility {
     }
 }
 
+/// Maximum number of allowed log viewers (specified in the interface spec).
+const MAX_ALLOWED_LOG_VIEWERS_COUNT: usize = 10;
+
+pub type BoundedAllowedViewers =
+    BoundedVec<MAX_ALLOWED_LOG_VIEWERS_COUNT, UNBOUNDED, UNBOUNDED, PrincipalId>;
+
+/// Log visibility for a canister.
+/// ```text
+/// variant {
+///    controllers;
+///    public;
+///    allowed_viewers: vec principal;
+/// }
+/// ```
+#[derive(Default, Clone, CandidType, Deserialize, Debug, PartialEq, Eq, EnumIter)]
+pub enum LogVisibilityV2 {
+    #[default]
+    #[serde(rename = "controllers")]
+    Controllers,
+    #[serde(rename = "public")]
+    Public,
+    #[serde(rename = "allowed_viewers")]
+    AllowedViewers(BoundedAllowedViewers),
+}
+
+// Ensure backward compatibility with the old `LogVisibility`.
+// `LogVisibilityV2` extends `LogVisibility` with an additional variant, `AllowedViewers`.
+// When decoding, if `AllowedViewers` is encountered, it is changed to a default value
+// to maintain compatibility with the old `LogVisibility`.
+impl Payload<'_> for LogVisibilityV2 {
+    fn decode(blob: &'_ [u8]) -> Result<Self, UserError> {
+        match Decode!([decoder_config()]; blob, Self).map_err(candid_error_to_user_error) {
+            // Fall back to the default value.
+            Ok(Self::AllowedViewers(_)) => Ok(Self::default()),
+            Ok(log_visibility_v2) => Ok(log_visibility_v2),
+            Err(err) => Err(err),
+        }
+    }
+}
+
+impl From<LogVisibility> for LogVisibilityV2 {
+    fn from(item: LogVisibility) -> Self {
+        match item {
+            LogVisibility::Controllers => Self::Controllers,
+            LogVisibility::Public => Self::Public,
+        }
+    }
+}
```

Comment:
I think this not used anywhere anymore, so can we remove?

Reply:
This conversion is needed for writing checkpoint proto from the `LogVisibility` v1 that we currently have and it has to be saved both to v1 and v2.